src/appshell/view/preferences/appearancepreferencesmodel.cpp

```diff
@@ -78,6 +93,51 @@ QStringList AppearancePreferencesModel::accentColors() const
     return uiConfiguration()->possibleAccentColors();
 }
 
+void AppearancePreferencesModel::resetThemeToDefault()
+{
+    uiConfiguration()->resetCurrentThemeToDefault(currentTheme().codeKey);
+    notationConfiguration()->resetCurrentBackgroundColorToDefault();
+    emit backgroundColorChanged();
+    emit themesChanged();
+}
+
+bool AppearancePreferencesModel::enableHighContrastChecked()
+{
+    if (isCurrentThemeHighContrast()) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void AppearancePreferencesModel::loadLastUsedGeneralTheme()
+{
+    uiConfiguration()->loadLastUsedGeneralTheme();
+}
+
+void AppearancePreferencesModel::loadLastUsedHighContrastTheme()
+{
+    uiConfiguration()->loadLastUsedHighContrastTheme();
+}
+
+void AppearancePreferencesModel::setNewColor(const QColor& newColor, const QString& propertyName)
+{
+    //! NOTE: Considered using a "switch()" statement here, but it would require a type conversion
+    //! from std::string to some form of a primitive literal. This has a workaround by implementing
+    //! a hash function, but I went for the "if/else if" ladder instead since we don't have that many cases anyway.
+
+    if (propertyName.toStdString() == "Accent Color:") {
```

Comment:
Better to create some enum and use it instead of text from UI

Reply:
I apologize, for i don't know how to properly implement an enum in this case. But i understand why using text from UI is just bad practice.

Should i create a new `propertyKey` variable in `UiColorsSection.qml` like so:
```
model: [
                { textRole: "Accent Color:", colorRole: ui.theme.accentColor, propertyKey: 0 },
                { textRole: "Text and Icons:", colorRole: ui.theme.fontPrimaryColor, propertyKey: 1},
                { textRole: "Disabled Text:", colorRole: "#000000", propertyKey: 2},
                { textRole: "Border Color:", colorRole: ui.theme.strokeColor, propertyKey: 3}
            ]
```

and then change the if/else-if ladder in `appearancepreferencesmodel.cpp` to a `switch()` statement like so:
```
switch(propertyKey)
{
    case 0: uiConfiguration()->setCurrentThemeStyleValue(ThemeStyleKey::ACCENT_COLOR, Val(newColor)); 
        break;
    
    case 1: uiConfiguration()->setCurrentThemeStyleValue(ThemeStyleKey::FONT_PRIMARY_COLOR, Val(newColor)); 
        break;
    
    case 2: return; //because a common way to refer to disabled items/text isn't fully implemented yet
    
    case 3: uiConfiguration()->setCurrentThemeStyleValue(ThemeStyleKey::STROKE_COLOR, Val(newColor)); 
        break;

    default: qDebug("Setting not Found!");    
```


Again, I'm suggesting this because i do not fully understand the enum approach.

Reply:
You can declare an enum in appearancepreferencesmodel.h (see navigationpanel.h for example):

<img width="778" alt="Screenshot 2021-08-03 at 18 15 30" src="https://user-images.githubusercontent.com/67867444/128050700-0f02581b-329f-4331-a46c-9e2fcfea6ff5.png">

Like this:

ColorType {
AccentColor,
TextAndIconsColor,
...
}
Q_ENUM(ColorType)

You can use this enum in both qml and cpp

Reply:
So, your function will be like this:

Q_INVOKABLE void setNewColor(const QColor& newColor, ColorType colorType)

Reply:
And in QML you access the enum values like 
```qml
AppearancePreferencesModel.AccentColor
AppearancePreferencesModel.TextAndIconsColor
```
You need to use AppearancePreferencesModel instead of the name of the enum in C++. This is by the design of Qt.

Reply:
Understood!
Enum implemented, and is working properly :).
Thank you for the tip Casper!